### DISCUSSION TITLE
request.validate not raise error in test

### DISCUSSION THREAD

**Discussion by crisnamurti** [adonisjs/core#4173] [Help] [answered]
Hi, the request.validate does not raise an exception when i send a invalid form on tests. If i remove await from request.validate its works.

export default class SignupController {
    public async handle({ request }: HttpContextContract) {
        const signupSchema = schema.create({
            email: schema.string({ trim: true }, [
                rules.email(),
            ]),
            firstName: schema.string({ trim: true }, [
                rules.minLength(3)
            ]),
            lastName: schema.string({ trim: true }, [
                rules.minLength(3)
            ]),
            password: schema.string({ trim: true }, [
                rules.confirmed('password2'),
                rules.regex(/^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[^\w\s]).{8,}$/),
            ]),
        })
        await request.validate({
            schema: signupSchema
        })

        return "success"
    }
}

And the tests:

  test('test user register', async ({ client }) => {
    const response = await client
      .post('signup')
      .form({})
      .withCsrfToken()

    response.assertStatus(200)
    response.dumpBody()
  })

**Comment by MaximeMRF** [comment, marked as answer]
Hello @crisnamurti 

If you remove `await` of your `request.validate`, node doesn't wait for the response of `request.validate` and pass automatically at the next line. Concretly it doesn't validate your request.

When you provide bad datas at your validator, `request.validate` will respond with a http 422 error with the reasons of the validation fail, but not raise an exeption.

**Reply by crisnamurti** [reply]
Thanks for you help. But it's a little weird.
I put a try catch and i get an exception ValidationException with flashmessages, in the browser application return back to signup form and i show the message validator, but in tests its not redirect to form its redirect me to root route, and flashmessages dont reach me.
Sorry about my english.

try {
    await request.validate({schema: signupSchema})
} catch(error) {
   console.log(error)
   throw error
}
